File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/vectorstore-similarity-search.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
from langchain_core.documents import Document
from langchain_core.vectorstores import InMemoryVectorStore

store = InMemoryVectorStore(openai_embeddings())
store.add_documents(
    [
        Document(page_content="Parity document one.", id="doc-1", metadata={"source": "one.txt"}),
        Document(page_content="Parity document two.", id="doc-2", metadata={"source": "two.txt"}),
    ]
)
store.similarity_search("parity", k=2)
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/vectorstore-similarity-search.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const { MemoryVectorStore } = helpers.classicModule('@langchain/classic/vectorstores/memory')
  const { Document } = helpers.coreModule('@langchain/core/documents')
  const store = new MemoryVectorStore(helpers.openAIEmbeddings())
  await store.addDocuments([
    new Document({ pageContent: 'Parity document one.', id: 'doc-1', metadata: { source: 'one.txt' } }),
    new Document({ pageContent: 'Parity document two.', id: 'doc-2', metadata: { source: 'two.txt' } }),
  ])
  await store.similaritySearch('parity', 2)
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})
